test/gui/shared/steps/file_context.py

```diff
@@ -123,12 +128,16 @@ def step(context, username, filename, filesize):
 
 @When(r'the user copies the (file|folder) "([^"]*)" to "([^"]*)"', regexp=True)
 def step(context, resource_type, source_dir, destination_dir):
-    source_dir = getResourcePath(source_dir)
-    destination_dir = getResourcePath(destination_dir)
+    source_dir_path = getResourcePath(source_dir)
+    destination_dir_path = getResourcePath(destination_dir)
     if resource_type == 'folder':
-        return shutil.copytree(source_dir, destination_dir)
+        return shutil.copytree(source_dir_path, destination_dir_path)
     else:
-        return shutil.copy2(source_dir, destination_dir)
+        try:
+            return shutil.copy2(source_dir_path, destination_dir_path)
+        except shutil.SameFileError:
+            copy_of_source_dir = addCopySuffix(source_dir_path)
+            return shutil.copy2(source_dir_path, copy_of_source_dir)
```

Comment:
I have another thought now. how about this flow:
- check if `source` == `destination`
- if same
  -  append destination with necessary suffix
  - do the copy
- if not
  - do normal copy

can be done for both file and folder. Then we don't have to use try-except